manifests/init.pp

```diff
@@ -128,15 +134,17 @@
   Hash                           $clients_config_hash,
   Hash                           $positions_config_hash,
   Hash                           $scrape_configs_hash,
-  Stdlib::Absolutepath           $bin_dir,
+  Stdlib::Absolutepath           $bin_dir               = promtail::params::bin_dir,
```

Comment:
my personal opinion is that instead of adding a params.pp, hiera in modules should be used (but I'm not a maintainer here).

Reply:
Agreed, please move this to the existing in module data. I will comment below with some pointers in case this is not something you have done before (no disrespect intended if you have, just trying to help)

Reply:
Would be nice. If you could do it. Also would be nice if you could have a look at the windows service thing, it seems to not work correctly.